packages/components/src/__actions__/Button/v3/Button.tsx

```diff
@@ -31,14 +42,24 @@ export const Button = ({
     <RACButton
       className={mergeClassNames(
         styles.button,
-        styles[variant],
         styles[size],
-        isReversed && styles.reversed,
+        isDisabled && styles.isDisabled,
+        isReversed ? styles[`${variant}Reversed`] : styles[variant],
+        isFullWidth && styles.fullWidth,
         className
       )}
+      isDisabled={isDisabled}
       {...otherProps}
     >
-      {children}
+      {racStateProps => {
+        const childIsFunction = typeof children === "function"
+
+        return (
+          <ButtonContent size={size} {...otherProps}>
```

Comment:
You're spreading `otherProps` both here and in the RACButton, which we shouldn't do because it duplicates props which either create conflict (eg. defining `data-testid`) or may be incompatible with one or the other.
Given it's more useful to put such attributes on the outer container and you're extending the library props, I'd leave it on the outer one and update the props in here to be explicit definitions.